pkg/controller/jobframework/validation.go

```diff
@@ -83,3 +84,13 @@ func ValidateUpdateForParentWorkload(oldJob, newJob GenericJob) field.ErrorList
 	}
 	return allErrs
 }
+
+func ValidateUpdateForWorkloadPriorityClassName(oldJob, newJob GenericJob) field.ErrorList {
+	var allErrs field.ErrorList
+	oldName, _ := WorkloadPriorityClassName(oldJob)
+	newName, _ := WorkloadPriorityClassName(newJob)
+	if oldName != newName {
+		allErrs = append(allErrs, field.Forbidden(workloadPriorityClassNamePath, "must not update workloadPriorityClass name"))
+	}
```

Comment:
Do you mean like this?
`allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldName, newName, workloadPriorityClassNamePath)...)`

BTW, in validation.go, other functions use `field.Forbidden`. 
How do we differentiate between `field.Forbidden` and `apivalidation.ValidateImmutableField` ?

Reply:
> Do you mean like this?
allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldName, newName, workloadPriorityClassNamePath)...)

Yes, I meant what you say.

> BTW, in validation.go, other functions use field.Forbidden.
How do we differentiate between field.Forbidden and apivalidation.ValidateImmutableField ?

Uhm, I see. Although, I think we should use `apivalidation.ValidateImmutableField` in the below places:

https://github.com/kubernetes-sigs/kueue/blob/91d5fa44b6b06c15778700f7126578816de3cb79/pkg/controller/jobframework/validation.go#L60-L68

https://github.com/kubernetes-sigs/kueue/blob/91d5fa44b6b06c15778700f7126578816de3cb79/pkg/controller/jobframework/validation.go#L70-L76

and here.

@mimowo Do you remember why you use `field.Forbidden` instead of `apivalidation.ValidateImmutableField` in 
[pkg/controller/jobframework/validation.go](https://github.com/kubernetes-sigs/kueue/pull/633/files#diff-5c2c085d3754c50f6e899f937ef8113a2fc81685d7130b5bf88d1a37d2f26d67): https://github.com/kubernetes-sigs/kueue/pull/633?

Reply:
Thanks. Fixed it

Reply:
it probably just slipped. We should prefer to use `ValidateImmutableField` to reduce code repetion.

Reply:
Sure, will create an issue.